core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -372,6 +372,13 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   final def eventually(implicit ev: CanFail[E]): URIO[R, A] =
     self orElse eventually
 
+  /**
+   * Executes this effect and returns its value, if it succeeds, but otherwise
+   * returns the specified value.
+   */
+  final def fallback[A1 >: A](a: => A1)(implicit ev: CanFail[E]): ZIO[R, Nothing, A1] =
+    fold(_ => a, identity)
```

Comment:
`catchAll(_ => a)`

Reply:
I'm not sure this is better if we need to wrap the result in an additional effect constructor.

Reply:
There’s no difference in that regard, one additional inspection in `fold` in fact. But it’s a shame it’s longer, perhaps there should have been a `catchAllM` too.

Reply:
You're right. Aren't they the the same though? Both are implemented in terms of `foldM` and have to wrap the success or failure in an effect constructor. `fold` does both of them internally whereas `catchAll` only does the success case itself and then we would need to wrap the failure case in an effect constructor here.

Reply:
Yep, they are. You must be getting tired of hearing this, but: awesome work 😅

Reply:
Thanks! Lot of people including you doing awesome work on this project! Just trying to contribute and learn. 😃